api/plugin_test.go

```diff
@@ -0,0 +1,702 @@
+		s.ctx.Params = gin.Params{{Key: "id", Value: "1"}}
+		s.a.DisablePlugin(s.ctx)
+
+		assert.Equal(s.T(), 500, s.recorder.Code)
+
+		assert.False(s.T(), s.db.GetPluginConfByUserAndPath(1, mock.ModulePath).Enabled)
+		s.resetRecorder()
+	}
+
+}
+
+func (s *PluginSuite) Test_EnableDisablePlugin_incorrectUser() {
+	{
+		test.WithUser(s.ctx, 2)
+
+		s.ctx.Request = httptest.NewRequest("POST", "/plugin/1/enable", nil)
+		s.ctx.Params = gin.Params{{Key: "id", Value: "1"}}
+		s.a.EnablePlugin(s.ctx)
+
+		assert.Equal(s.T(), 401, s.recorder.Code)
+		log.Println(s.recorder.Body.String())
+
+		assert.False(s.T(), s.db.GetPluginConfByUserAndPath(1, mock.ModulePath).Enabled)
+		s.resetRecorder()
+	}
+
+	{
+		test.WithUser(s.ctx, 2)
+
+		s.ctx.Request = httptest.NewRequest("POST", "/plugin/1/disable", nil)
+		s.ctx.Params = gin.Params{{Key: "id", Value: "1"}}
+		s.a.DisablePlugin(s.ctx)
+
+		assert.Equal(s.T(), 401, s.recorder.Code)
+
+		assert.False(s.T(), s.db.GetPluginConfByUserAndPath(1, mock.ModulePath).Enabled)
+		s.resetRecorder()
+	}
+
+}
+
+func (s *PluginSuite) Test_EnableDisablePlugin_nonExistPlugin() {
+	{
+		test.WithUser(s.ctx, 1)
+
+		s.ctx.Request = httptest.NewRequest("POST", "/plugin/99/enable", nil)
+		s.ctx.Params = gin.Params{{Key: "id", Value: "99"}}
+		s.a.EnablePlugin(s.ctx)
+
+		assert.Equal(s.T(), 404, s.recorder.Code)
+		s.resetRecorder()
+	}
+
+	{
+		test.WithUser(s.ctx, 1)
+
+		s.ctx.Request = httptest.NewRequest("POST", "/plugin/99/disable", nil)
+		s.ctx.Params = gin.Params{{Key: "id", Value: "99"}}
+		s.a.DisablePlugin(s.ctx)
+
+		assert.Equal(s.T(), 404, s.recorder.Code)
+		s.resetRecorder()
+	}
+
+}
+
+func (s *PluginSuite) Test_EnableDisablePlugin_danglingConf() {
+	conf := s.getDanglingConf(1)
+
+	{
+		test.WithUser(s.ctx, 1)
+
+		s.ctx.Request = httptest.NewRequest("POST", fmt.Sprintf("/plugin/%d/enable", conf.ID), nil)
+		s.ctx.Params = gin.Params{{Key: "id", Value: fmt.Sprint(conf.ID)}}
+		s.a.EnablePlugin(s.ctx)
+
+		assert.Equal(s.T(), 404, s.recorder.Code)
+		s.resetRecorder()
+	}
+
+	{
+		test.WithUser(s.ctx, 1)
+
+		s.ctx.Request = httptest.NewRequest("POST", fmt.Sprintf("/plugin/%d/disable", conf.ID), nil)
+		s.ctx.Params = gin.Params{{Key: "id", Value: fmt.Sprint(conf.ID)}}
+		s.a.DisablePlugin(s.ctx)
+
+		assert.Equal(s.T(), 404, s.recorder.Code)
+		s.resetRecorder()
+	}
+}
+
+func (s *PluginSuite) Test_GetDisplay() {
+	conf := s.db.GetPluginConfByUserAndPath(1, mock.ModulePath)
+	inst, err := s.manager.Instance(conf.ID)
+	assert.Nil(s.T(), err)
+	mockInst := inst.(*mock.PluginInstance)
+
+	mockInst.DisplayString = "test string"
+
+	{
+		test.WithUser(s.ctx, 1)
+
+		s.ctx.Request = httptest.NewRequest("GET", fmt.Sprintf("/plugin/%d/display", conf.ID), nil)
+		s.ctx.Params = gin.Params{{Key: "id", Value: fmt.Sprint(conf.ID)}}
+		s.a.GetDisplay(s.ctx)
+
+		assert.Equal(s.T(), 200, s.recorder.Code)
+		test.JSONEquals(s.T(), mockInst.DisplayString, s.recorder.Body.String())
```

Comment:
I think since it is still an API, returning JSON for everything is good for keeping things consistent as escaping some unwanted values which might cause issues during transfer(maybe?). `axios` will parse JSON automatically by its `content-type`